### ISSUE TITLE
Glyco Search

### ISSUE THREAD

**Issue by fazeliniah** [smith-chem-wisc/MetaMorpheus#2019]
Hi MetaMorpheus developers, 
Is there any tutorial for running the MM tool for o-linked GlycoProteomics? I tried the MM for few o-linked enriched raw files (HCD fragmentation) using two tasks: 1- Calibration and 2- Glycosearch.
Some quick questions: 
1- Is MM able to perform XIC and report the intensity values for Glyco peptides? 
2- We would like to validate the findings by checking the spectra. How can we visualize oxonium ions in MetaDraw?
Thanks

**Comment by lonelu** [comment]
Hi
Thanks for using MetaMorpheus.  You can find the tutorial:
https://github.com/smith-chem-wisc/MetaMorpheus/wiki/1_New-Task:-O-Glyco-Search
https://drive.google.com/drive/folders/1_xAcH1k2Bqf-IyTAsUsK_WuM9o0qKj3Y

1. You can use Calibration before GlycoSearch, but I don't suggest it now as we haven't tested Calibration with GLycosearch.
2. Quantification can be a good feature. We don't report intensity in the current MM GlycoSearch. However, I have a tested version that could report intensity. Let me know if you are interested.
3. I think you can visualize all matched ions include oxonium ions in MetaDraw. Basically, you just need to add a raw file and the corresponding psms file into MetaDraw, it will show you the matches.

**Comment by fazeliniah** [comment]
Thank you so much your help and very useful tutorial. 
I have couple of questions and greatly appreciate your help. 
1- It seems that the GlycoSearch does much better/faster when the number of proteins in the FASTA file is small. What is your recommendation for larger databases (e.g. entire mouse proteome).  Is there any recommendation to generate focused FASTS protein database using MetaMorpheus?
2- Is there a way to reduce the font size for peptide sequence MetaDraw window? This will be useful for larger peptide. 
Thanks again for this great software.

**Comment by lonelu** [comment]
I agree with you that the current version performs better with a small database. We are still working on improving the scoring function to remove the database size effect. I assume this improvement will be finished in the near future and I will let you know.

It is possible to reduce the size of the databases. You can perform a normal search with the 'Search Task'. And check 'Search Task' -> Advanced Options -> Post-Search Analysis -> Write Two Pruned Databases. After the normal search, you will get a focused database. You can then use the focused database to perform a Glycosearch. However, this may depend on the data. We do have users report crashes. Let me know if this is not working for you.

The font size for the peptide sequence in MetaDraw window is not changeable yet. I will discuss with Shortreed to see how to make it changeable. 
Best,
Lei

**Comment by fazeliniah** [comment]
I generated the focused database but unfortunately (as you suggested) the Glycosearch failed with "a Runtime Error: "Run failed, Exception: Index was outside the bounds of the array".  I used a customized .fasta file (generated manually) and Glycosearch worked with no error.
